### ISSUE TITLE
[BUG] Zero downtime upgrade stuck in "Virt launcher pod is terminating" caused by VM crashed (4 nodes)

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#1717]
**Describe the bug**

Perform zero downtime upgrade on 4 nodes harvester from v1.0.0-rc2 to customized release. 

> Upgraded VM crashed during the process which cause node provisioning stuck

* Upgrade usage vm crash 
![image](https://user-images.githubusercontent.com/29251855/146798593-84b29a71-84fb-4b29-a53d-3a062cd65618.png)

* Virt launcher pod is terminating
![image](https://user-images.githubusercontent.com/29251855/146798534-a90718d5-afc5-4793-a6f8-e9556c0410ec.png)

* Upgrade process keep stuck in node 3
![image](https://user-images.githubusercontent.com/29251855/146798438-eecc588d-1c5e-48d6-be6a-3f2cd69832cb.png)

**To Reproduce**
1. Provide 4 nodes harvester cluster under air gap network environment
2. Create image from upload file 
3. Create a 2 core, 4GB ubuntu vm 
4. Enable harvester mgmt in virtual network
5. Create a virtual network `vlan1` with id `1`
6. Prepare zero downtime upgrade ISO file with bumped rke2 and rancher version
7. Open harvester settings
8. Open the `Upgrade` beside the server-version 
9. Upload ISO file 
10. Provide sha512 value 
11. Click Upgrade button 
12. Check upgrade status

**Expected behavior**

* Should do zero downtime upgrade correctly and upgrade harvester, os, rke2 and rancher to latest version in ISO
* Upgrade usage vm should not crashed due to I/O low response timeout

**Support bundle**

[supportbundle_6ef2a384-9005-4692-a4d7-c7ac911e69ed_2021-12-20T13-24-43Z.zip](https://github.com/harvester/harvester/files/7746633/supportbundle_6ef2a384-9005-4692-a4d7-c7ac911e69ed_2021-12-20T13-24-43Z.zip)

**Environment:**
 - Harvester ISO version: v1.0.0-rc2
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 4 nodes local kvm machines

**Additional context**

**Comment by bk201** [comment]
Seeing two issues:
- upgrade-repo VM is terminated because the readiness probe fails. I guess it's because of heavy loading on the test env. 
  - The VM is stuck at terminating. Normally, the VM should be restarted and continue serving.


     ```
     harvester-node-0:~ # kubectl get pods -n harvester-system | grep virt-l
     virt-launcher-upgrade-repo-hvst-upgrade-rplg5-5mxc7       0/1     Terminating   0               44m
     
     harvester-node-0:~ # kubectl get vmi -n harvester-system
     NAME                              AGE   PHASE    IP            NODENAME           READY
     upgrade-repo-hvst-upgrade-rplg5   99m   Failed   10.52.0.128   harvester-node-0   False
     
     harvester-node-0:~ # kubectl describe pod -n harvester-system virt-launcher-upgrade-repo-hvst-upgrade-rplg5-5mxc7
     
     ...
       Normal   Pulled                        44m                kubelet                  Container image "registry.suse.com/suse/sles/15.3/virt-launcher:0.45.0-8.4.3" already present on machine
       Normal   Created                       44m                kubelet                  Created container compute
       Normal   Started                       44m                kubelet                  Started container compute
       Warning  Unhealthy                     44m                kubelet                  Readiness probe failed: Get "http://10.52.0.128:80/harvester-iso/harvester-release.yaml": dial tcp 10.52.0.128:80: i/o timeout (Client.Timeout exceeded while awaiting headers)
       Warning  Unhealthy                     44m                kubelet                  Readiness probe failed: Get "http://10.52.0.128:80/harvester-iso/harvester-release.yaml": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
       Warning  MultiplePodDisruptionBudgets  43m (x6 over 45m)  controllermanager        Pod "harvester-system"/"virt-launcher-upgrade-repo-hvst-upgrade-rplg5-5mxc7" matches multiple PodDisruptionBudgets.  Chose "kubevirt-disruption-budget-2c4bt" arbitrarily.
       Nor
  - Tuning the timeout value in https://github.com/harvester/harvester/pull/1716
- Seeing regression of pre-drain/post-drain hooks issue as spotted in https://github.com/rancher/rancher/issues/35464#issuecomment-976633077, multiple nodes are cordoned at once.

     ```
     harvester-node-0:~ # kubectl get nodes
     NAME               STATUS                     ROLES                       AGE     VERSION
     harvester-node-0   Ready                      control-plane,etcd,master   8h      v1.22.4+rke2r2
     harvester-node-1   Ready                      control-plane,etcd,master   7h56m   v1.22.4+rke2r2
     harvester-node-2   Ready,SchedulingDisabled   control-plane,etcd,master   7h43m   v1.22.4+rke2r2
     harvester-node-3   Ready,SchedulingDisabled   <none>                      7h27m   v1.22.4+rke2r2
     ```

**Comment by bk201** [comment]
This issue is currently blocked by https://github.com/rancher/rancher/issues/36536, the implementation in https://github.com/rancher/rancher/issues/35999 can't be verified yet.

**Comment by TachunLin** [comment]
Verified **fixed** on **`v1.0.1-rc3`**. Close this issue.

### Result
* Can 100% upgrade 4 nodes harvester (3 management, 1 agent node) from v1.0.0 to v1.0.1-rc3
  ![image](https://user-images.githubusercontent.com/29251855/162408820-e2540958-f436-48e3-a7b1-31992b788e7c.png)
* Did not encounter multiple nodes upgrades at the same time or cordoned at once. Upgrade node are running one by one
 ![image](https://user-images.githubusercontent.com/29251855/162402046-23ee4fca-da73-43ca-8fef-e669b20670b9.png)
 ![image](https://user-images.githubusercontent.com/29251855/162403597-59e5f44c-cb05-4ad4-8245-30bd1ac20c42.png)
  

### Test Information
* Test Environment: 4 nodes harvester on dolphin bare machines
* Harvester version: v1.0.0 GA to v1.0.1-rc3

### Verify Steps
1. Prepare v1.0.0 four nodes harvester cluster on Provo machines
2. Create a ubuntu image
3. Create a ubuntu virtual machine
4. Enable network to `harvester-mgmt`
5. Shutdown all VMs
6. Follow the steps to upgrade to v1.0.1-rc3
https://github.com/harvester/docs/blob/42598035cfe79856f7e159d4c47c4d7d29178900/docs/upgrade/automatic.md
7. Check each node status during `Upgrade Node` process 

### Additional Context
* Support Bundle
[supportbundle_800060cb-051a-47cd-8bff-7cb1ae5090aa_2022-04-08T10-17-57Z.zip](https://github.com/harvester/harvester/files/8450956/supportbundle_800060cb-051a-47cd-8bff-7cb1ae5090aa_2022-04-08T10-17-57Z.zip)
